src/raft/util.go

```diff
@@ -1,13 +1,36 @@
 package raft
 
-import "log"
+import (
+	"labrpc"
+	"log"
+	"math/rand"
+	"time"
+)
 
 // Debugging
-const Debug = 0
+const Debug = 1
 
 func DPrintf(format string, a ...interface{}) (n int, err error) {
 	if Debug > 0 {
 		log.Printf(format, a...)
 	}
 	return
 }
+
+func randomTimeout(minVal time.Duration) <-chan time.Time {
+	if minVal == 0 {
+		return nil
+	}
+	extra := (time.Duration(rand.Int63()) % minVal)
+	return time.After(minVal + extra)
+}
```

Comment:
This method returns a timer that sends a message on a channel after the timeout. Nicely, this fits right into the select statements above where we wait for timeouts or messages.